sub/sd_ass.c

```diff
@@ -239,11 +243,16 @@ static void decode(struct sd *sd, struct demux_packet *packet)
     struct sd_ass_priv *ctx = sd->priv;
     ASS_Track *track = ctx->ass_track;
     if (ctx->converter) {
-        if (!sd->opts->sub_clear_on_seek && check_packet_seen(sd, packet->pos))
+        if (!sd->opts->sub_clear_on_seek && packet->pos >= 0 && check_packet_seen(sd, packet->pos))
             return;
         char **r = lavc_conv_decode(ctx->converter, packet);
         for (int n = 0; r && r[n]; n++)
             ass_process_data(track, r[n], strlen(r[n]));
+        if (ctx->duration_unknown && r && *r) {
+            for (int n = 0; n < track->n_events-1; n++) {
+                track->events[n].Duration = track->events[n+1].Start - track->events[n].Start;
+            }
+        }
```

Comment:
Probably better to always run this loop, instead of only under a stragne looking condition (``r && *r``)? Or can the CC decoder return "nothing" for meaning that the current subtitle state doesn't change? (IMO this is getting strange - so returning an empty string would mean terminate the current subtitle event, and returning "nothing" means keep it?)

Cosmetic complaint: lines should be wrapped to 80 columns, and also the coding style is not always consistent, e.g. ``n+1`` should be ``n + 1``.